pytests/test_operators.py

```diff
@@ -379,24 +368,31 @@ def trigger(item):
 
     flow.flat_map(trigger)
 
-    clock_config = TestingClockConfig(clock)
+    clock_config = TestingClockConfig(lambda e: e["time"], wait_until_end=False)
```

Comment:
What is your thinking behind using the `TestingClockConfig` here rather than the `EventTimeClockConfig`? My thinking was that the operator functionality can still be tested using event time and so we should use that clock directly to make it clear that is what's happending. We'll save the `TestingClock` as is for very special (and confusing) cases where we need to mock system time, which I don't think we have to do for any of these tests.

Reply:
So, the `TestingClockConfig` is technically an `EventClockConfig` with a fixed configuration.
Since we are not testing the clock, but the operators (and/or recovery), we do this by using a clock that can represent the two use cases we have in tests:
- We want to wait untill the end of the input before emitting anything
- We want to emit anything as soon as possible (to test for recovery)

We could do that with a normal `EventclockConfig`, but then we would have to be aware of never using it with a `wait_for_system_duration` that is not `0` or `longer than the timespan of all event times`.

So, I removed the old TestingClock since the idea of this PR was to move all tests to use an event time clock, and created a specific event time clock that only allows the behaviours we are currently using in tests.

This way we avoid giving the possibility of partially mocking system time (which creates more problems than it solves), and we can use event time without worrying about flaky behaviour due to system timing.